### ISSUE TITLE
[Question] Is there a way to parallelise PPO so that it can run on GPU?  

### ISSUE THREAD

**Issue by Dylan-Kerler** [DLR-RM/stable-baselines3#374]
### Question

The main bottleneck of PPO is the env simulation; A gradient update cannot be done until enough "experience" is collected. Is it possible to run the env simulation in parallel and then asynchronously do gradient updates?

**Comment by araffin** [comment]
Probably a duplicate of https://github.com/DLR-RM/stable-baselines3/issues/314
You may also take a look at MPI, and would accept a PR in contrib repo, see https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/issues/11

**Comment by Dylan-Kerler** [comment]
I had never even heard of MPI before you mentioned it (i looked into it and it's super cool).
